src/test/java/ch/uzh/ifi/hase/soprafs22/screwyourneighborserver/api/CardApiTest.java

```diff
@@ -0,0 +1,160 @@
+            .body(BodyInserters.fromValue(PLAYER_2))
+            .exchange()
+            .expectStatus()
+            .isCreated()
+            .expectBody()
+            .returnResult()
+            .getResponseHeaders();
+
+    String sessionIdPlayer2 = getSessionIdOf(responseHeaders);
+
+    Player player1 =
+        playerRepository.findAll().stream()
+            .filter(player -> player.getName().equals(PLAYER_NAME_1))
+            .findFirst()
+            .orElseThrow();
+
+    Player player2 =
+        playerRepository.findAll().stream()
+            .filter(player -> player.getName().equals(PLAYER_NAME_2))
+            .findFirst()
+            .orElseThrow();
+
+    Game game =
+        GameBuilder.builder("game1", gameRepository, participationRepository, playerRepository)
+            .withParticipationWith(player1)
+            .withParticipationWith(player2)
+            .withGameState(GameState.PLAYING)
+            .withMatch()
+            .withMatchState(MatchState.ANNOUNCING)
+            .withHandForPlayer(PLAYER_NAME_1)
+            .withCards(ACE_OF_CLUBS, QUEEN_OF_CLUBS)
+            .finishHand()
+            .withHandForPlayer(PLAYER_NAME_2)
+            .withCards(KING_OF_CLUBS, JACK_OF_CLUBS)
+            .finishHand()
+            .withRound()
+            .withPlayedCard(PLAYER_NAME_1, ACE_OF_CLUBS)
+            .finishRound()
+            .finishMatch()
+            .build();
+
+    gameRepository.saveAll(List.of(game));
+
+    String uri = "games/" + game.getId();
+
+    webTestClient
+        .get()
+        .uri(uri)
+        .header(HttpHeaders.COOKIE, "JSESSIONID=%s".formatted(sessionIdPlayer1))
+        .exchange()
+        .expectStatus()
+        .isOk()
+        .expectBody()
+        .jsonPath("_embedded.matches[*].hands[*].cards[*].cardRank")
+        .value(containsInAnyOrder(ACE.name(), QUEEN.name(), null, null))
+        .jsonPath("_embedded.matches[*].hands[*].cards[*].cardSuit")
+        .value(containsInAnyOrder(CLUB.name(), CLUB.name(), null, null))
+        .jsonPath("_embedded.matches[*].rounds[0].cards[0].cardRank")
+        .isEqualTo(ACE.name())
+        .jsonPath("_embedded.matches[*].rounds[0].cards[0].cardSuit")
+        .isEqualTo(CLUB.name());
+
+    webTestClient
+        .get()
+        .uri(uri)
+        .header(HttpHeaders.COOKIE, "JSESSIONID=%s".formatted(sessionIdPlayer2))
+        .exchange()
+        .expectStatus()
+        .isOk()
+        .expectBody()
+        .jsonPath("_embedded.matches[*].hands[*].cards[*].cardRank")
+        .value(containsInAnyOrder(ACE.name(), null, KING.name(), JACK.name()))
+        .jsonPath("_embedded.matches[*].hands[*].cards[*].cardSuit")
+        .value(containsInAnyOrder(CLUB.name(), null, CLUB.name(), CLUB.name()))
+        .jsonPath("_embedded.matches[*].rounds[0].cards[0].cardRank")
+        .isEqualTo(ACE.name())
+        .jsonPath("_embedded.matches[*].rounds[0].cards[0].cardSuit")
+        .isEqualTo(CLUB.name());
+  }
+}
```

Comment:
Hmm In my understanding this test does not show necessarily that the serializer is integrated. Because the test hide_correct_cards would also work with the old implementation, right? That's why, I'd prefer a test which would not work without the CardSerializer logic to show, that the overriden serialize method is really called.

Reply:
I pushed my proposal for a test, hide_own_card_in_match5(), if you're absolutely against it, don't hesitate to remove it again. At least I tested it for me and of course it worked everything as expected!! (how could I ever doubt that :P). 
Btw. I did not move the duplicate code of player creation out to the setup method, because I expect more tests being placed here in relation to the card validations, and maybe these tests need another setup  - I don't know yet.

Reply:
you can test it, the test should not work when the serializer is not overridden, because then you have no null values for CardRank or CardSuit

Reply:
aah ok :)

Reply:
i removed it again, sorry